contrib/gitian-descriptors/gitian-osx.yml

```diff
@@ -132,6 +133,31 @@ script: |
 
     make osx_volname
     make deploydir
+
+    # Hack to work around differences in the behavior of native_cctools' codesign_allocate and Apple distributed codesign and codesign_allocate.
+    # Specifically, the signature is created by appending it to the end of the binary and then updating the size for the section __LINKEDIT
+    # The "vmsize" attribute of this section is rounded to 0x1000, 0x2000, or 0x4000 depending on the architecture. For some reason, this also happens
+    # to differ between what we can build ourselves in native_cctools (which uses Apple's published source code) and what Apple distributes
+    # as "Xcode command line tools". It appears that the Apple distributed tool rounds to 0x2000.
+    #
+    # Fortunately codesign_allocate will not reduce the "vmsize" attribute, and when the signature is appended, the difference in actual size and
+    # "vmsize" will not be present in the final binary.
+    #
+    # Thus we can workaround this by calculating an overestimate of what the signature will be and then using codesign_allocate to preallocate
+    # space for the code signature. Because the same codesign_allocate is used in this step, the rounding problem will not be present.
+    # Since the space allocated for the signature should be larger than actually needed, the signing step will not reduce "vmsize" so we
+    # won't have any problems there.
+    #
+    # From inspecting previous releases' code signatures, we find that the code signature primarily consists of SHA256 hashes of 4096 byte chunk
+    # of the unsigned binary. There are a few other hashes of something. Then the signature itself, which is ~9kb. Then ~10kb of additional overhead.
+    # We can thus produce an overestimate as follows: (((unsigned_size / 4096) + 1) * 32) + 50000. This gives us an expected number of SHA256 hashes
+    # and how large they will be. By adding 50k, we can include the signature, overhead, and still have room if Apple decides to cram more stuff
+    # into the code signature.
+    BINARY_RESULT="dist/Bitcoin-Qt.app/Contents/MacOS/Bitcoin-Qt"
+    UNSIGNED_SIZE=`stat -c %s ${BINARY_RESULT}`
```

Comment:
```
In gitian-osx.yml line 122:

  UNSIGNED_SIZE=`stat -c %s ${BINARY_RESULT}`

                ^---------------------------^ SC2006: Use $(...) notation instead of legacy backticked `...`.

Did you mean: 

  UNSIGNED_SIZE=$(stat -c %s ${BINARY_RESULT})

In gitian-osx.yml line 123:

  SIG_SIZE_EST=$(((((${UNSIGNED_SIZE} / 4096) + 1) * 32) + 50000))

                     ^--------------^ SC2004: $/${} is unnecessary on arithmetic variables.

For more information:

  https://www.shellcheck.net/wiki/SC2004 -- $/${} is unnecessary on arithmeti...

  https://www.shellcheck.net/wiki/SC2006 -- Use $(...) notation instead of le...

^---- failure generated from test/lint/lint-shell.sh

Reply:
Fixed.